samples: nrf9160: modem_shell: ppp: updates (#6636)
nrfconnect/sdk-nrf: samples/nrf9160/modem_shell/README.rst

Examples
--------

> * PPP network interface is setup/setdown automatically when LTE connection is up/down. Set the PPP network interface up manually:

Author: Changed.

  .. code-block:: console

     ppp up

* Set the PPP network interface down manually:

  .. code-block:: console

     ppp down

* Set the custom baudrate configuration for PPP UART (default: 115200):

  .. code-block:: console

     ppp uartconf --baudrate 921600

> * Set the rts_cts HW flow control configuration for PPP UART (default: none):

Reviewer: ```suggestion
* Set the ``rts_cts`` hardware flow control configuration for PPP UART (default: none):
```

   west build -p -b nrf9160dk_nrf9160_ns -- -DDTC_OVERLAY_FILE=ppp.overlay -DOVERLAY_CONFIG=overlay-ppp.conf

> After flashing, testing with the development kit and Linux can be done as follows for example:

Reviewer: ```suggestion
After programming the development kit, you can test it in the Linux environment as follows:
```

1. Connect the development kit to the computer using a USB cable.
   The development kit is assigned a ttyACM device (Linux).

> #. Create a serial connection to the development kit (/dev/ttyACM2) with a terminal (e.g. PuTTY).

Reviewer: ```suggestion
#. Open a serial connection to the development kit (/dev/ttyACM2) with a terminal (for example PuTTY).
```

#. Reset the development kit.

#. Observe in the terminal window that the MoSh starts with the PPP support.
   This is indicated by output similar to the following (there is also a lot of additional information about the LTE connection):

   .. code-block:: console

      Network registration status: searching
      Network registration status: Connected - home network
      Default PDN is active: starting PPP automatically
      PPP: started
      mosh:~$

> #. Type ``ppp uartconf`` command resulting in the following uart configuration:

Reviewer: ```suggestion
#. Enter command``ppp uartconf`` that results in the following UART configuration:
```

> #. Type ``ppp uartconf`` commands resulting for example following uart configuration:

Author: Changed.

   .. code-block:: console

      mosh:~$ ppp uartconf -r
      PPP uart configuration:
        baudrate:     921600
        flow control: RTS_CTS
        parity:       none
        data bits:    bits8
        stop bits:    bits1
      mosh:~$

> #. In Linux terminal start the ppp connection by typing the following:

Reviewer: ```suggestion
#. In a Linux terminal, enter the following command to start the PPP connection:
```

> #. In Linux terminal start the ppp connection by typing following:

Reviewer 2: ```suggestion
#. In Linux terminal start the ppp connection by typing the following:
```

Author: Changed.

   .. code-block:: console

      $ sudo pppd -detach /dev/ttyACM0 921600 noauth crtscts noccp novj nodeflate nobsdcomp local debug +ipv6 ipv6cp-use-ipaddr usepeerdns noipdefault defaultroute

> #. In MoSh terminal observe that ppp connection is created:

Reviewer: ```suggestion
#. In a MoSh terminal, observe that the PPP connection is created:
```

   .. code-block:: console

      Dial up (IPv6) connection up
      Dial up (IPv4) connection up
      mosh:~$

> #. Now you are ready to browse Internet in Linux by using setuped MoSh ppp dialup over LTE connection.

Reviewer: ```suggestion
Now, you are ready to browse Internet in Linux by using the MoSh PPP dial-up over LTE connection.
```
